src/Sema.zig

```diff
@@ -30738,15 +30738,33 @@ fn elemPtrType(sema: *Sema, ptr_ty: Type, offset: ?usize) !Type {
     const ptr_info = ptr_ty.ptrInfo().data;
     const elem_ty = ptr_ty.elemType2();
     const allow_zero = ptr_info.@"allowzero" and (offset orelse 0) == 0;
+    const target = sema.mod.getTarget();
+    const parent_ty = ptr_ty.childType();
+
+    const vector_info: struct {
+        host_size: u16,
+        bit_offset: u16,
+        alignment: u32,
+    } = if (parent_ty.tag() == .vector) blk: {
+        const elem_bits = elem_ty.bitSize(target);
+        const is_packed = (elem_bits & (elem_bits - 1)) != 0;
+        const is_packed_with_offset = is_packed and offset != null and offset.? != 0;
```

Comment:
Just a quick note: If I remember correctly, on big-endian systems the indexing order is reversed for LLVM vectors.

This reversal is so that the storage layout will match between `@Vector(N, T)` and `[N]T`  when T is a self-aligned (power-of-two) type

Reply:
how to run test command in big-endian emu ... 😊

Reply:
MIPS is a good target for big-endian.

You can add `-target mips-linux --test-cmd qemu-mips --test-cmd-bin` to the Zig invocation to test on MIPS, which is big-endian. On my Debian system, the packages to install to get QEMU working are "qemu-system-mips qemu-user"

Reply:
thx, and the result is as you said 😊